platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/NonAncientEventWindow.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.consensus;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.platform.event.GossipEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Determines the window of rounds between the pending consensus round and the minimum round non-ancient. Provides the
+ * following information:
+ * <ol>
+ *     <li> latestConsensusRound - the latest round to have come to consensus.</li>
+ *     <li> minRoundNonAncient - the ancient threshold based on event birth round</li>
+ *     <li> minGenNonAncient - the ancient threshold based on event generation</li>
+ *     <li> pendingConsensusRound - the current round coming to consensus, i.e. 1  + the latestConsensusRound</li>
+ * </ol>
+ * <p>
+ * FUTURE WORK: Remove minGenNonAncient once we throw the switch to using minRoundNonAncient as the ancient threshold.
+ */
+public record NonAncientEventWindow(long latestConsensusRound, long minRoundNonAncient, long minGenNonAncient) {
```

Comment:
Maybe in the future, this might not be a record, but a class that hides these variables. That way, we will prevent anyone from misusing these variables, they would have to go through the methods to determine if something is ancient.

Reply:
I will meditate on that.   By the end of this long refactor it may be as you say.   I'll add a tasklist for cleanup in the epic and link to this comment to come back to it.